### PR TITLE
opts/envfile: trim all leading whitespace in a line

### PR DESCRIPTION
**Pull request by mattrobenolt**
Rather than just trimming before a variable, we should trim the line first.

This prevents a line with whitespace only to not break docker.

### CODE REVIEW

**Comment by duglin** [comment]
testcases?

**Comment by mattrobenolt** [comment]
Yeah, adding now. :)

**Comment by mattrobenolt** [comment]
@duglin test added, which fails before this patch, but passes after patch.

**Comment by calavera** [comment]
LGTM

File: opts/envfile_test.go [hunk 1]
```diff
@@ -29,7 +29,12 @@ func TestParseEnvFileGoodFile(t *testing.T) {
 
 _foobar=foobaz
 `
-
+	// Adding a newline + a line with pure whitespace.
+	// This is being done like this instead of the block above
+	// because it's common for editors to trim trailing whitespace
+	// from lines, which becomes annoying since that's the
+	// exact thing we need to test.
+	content += "\n    "
```
Comment (thaJeztah) [inline opts/envfile_test.go]: Do we need a similar test, using a `tab` as whitespace?

Also, this looks to be adding *trailing* whitespace (the whitespace is appended to the content), not *leading* whitespace, is that correct?

Comment (thaJeztah) [inline opts/envfile_test.go]: think I misread the description of the PR, got confused because of the "trim the first line" in your PR description

Comment (mattrobenolt) [inline opts/envfile_test.go]: Ah, I see the misunderstanding, but it says "trim the line first" NOT "trim the first line". ;)

tl;dr, if there's a line with _only_ whitespace, docker would error before.

> Do we need a similar test, using a tab as whitespace?

I could just add a `\t` into this line as well to cover both.

Comment (thaJeztah) [inline opts/envfile_test.go]: Doh! You're right, probably combined with the "trim leading" title of the PR, I got completely lost LOL

> I could just add a \t into this line as well to cover both

Saw you added it, thanks!

**Comment by cpuguy83** [comment]
LGTM
